apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx

```diff
@@ -0,0 +1,164 @@
+import WorkshopEnrollmentPreSurvey from '@cdo/apps/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_survey';
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from 'chai';
+
+describe("WorkshopEnrollmentPreSurvey", () => {
+  describe("getSortableUnitLessonShortName()", () => {
+    let getSortableUnitLessonShortName;
+
+    before(() => {
+      getSortableUnitLessonShortName = shallow(
+        <WorkshopEnrollmentPreSurvey enrollments={[]}/>
+      ).instance().getSortableUnitLessonShortName;
+    });
+
+    it("correctly zero-pads the supplied unitLessonShortName", () => {
+      expect(getSortableUnitLessonShortName("U1 L1")).to.equal("U0001 L0001");
+      expect(getSortableUnitLessonShortName("U2 L10")).to.equal("U0002 L0010");
+    });
+
+    it("returns empty string if it can't parse the supplied string", () => {
+      expect(getSortableUnitLessonShortName("unparseable")).to.equal("");
+    });
+  });
+
+  describe("With Data", () => {
+    let userIndex = 0;
+    let fakeEnrollments;
+    let workshopEnrollmentPreSurvey;
+    let tableRows;
+    before(() => {
+      const generateFakeEnrollment = survey => {
+        userIndex++;
+        return {
+          first_name: `Teacher${userIndex}`,
+          last_name: `LastName${userIndex}`,
+          email: `teacher${userIndex}@ex.net`,
+          district_name: "A district",
+          school: "A school",
+          user_id: userIndex,
+          attended: true,
+          pre_workshop_survey: survey
+        };
+      };
+
+      fakeEnrollments = [
+        generateFakeEnrollment({
+          unit: "Unit 4 - the fouth unit",
+          lesson: "Lesson 3 - the third lesson",
+          questionsAndTopics: "so many questions",
+          unitLessonShortName: "U4 L3"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 1 - the first unit",
+          lesson: "Lesson 11 - the eleventh lesson",
+          questionsAndTopics: "another question",
+          unitLessonShortName: "U1 L11"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 1 - the first unit",
+          lesson: "Lesson 2 - the second lesson",
+          questionsAndTopics: "",
+          unitLessonShortName: "U1 L2"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 1 - the first unit",
+          lesson: "Lesson 1 - the first lesson",
+          questionsAndTopics: "",
+          unitLessonShortName: "U1 L1"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 4 - the fouth unit",
+          lesson: "Lesson 3 - the third lesson",
+          questionsAndTopics: "more questions...",
+          unitLessonShortName: "U4 L3"
+        }),
+        // one no response
```

Comment:
should that be `on no response`?

Reply:
No, I meant one enrollment with no response. How about that? `"One enrollment with no response"`? Or maybe it's self-explanatory enough without the comment?